moengage-segment-kotlin-destination/src/main/java/com/segment/analytics/kotlin/destinations/moengage/MoEngageDestination.kt

```diff
@@ -0,0 +1,242 @@
+data class MoEngageSettings(val apiKey: String)
+
+class MoEngageDestination(private val application: Application) : DestinationPlugin(),
+    VersionedPlugin {
+
+    private lateinit var integrationHelper: MoEIntegrationHelper
+    private lateinit var instanceId: String
+
+
+    companion object {
+        private const val tag = "MoEngageDestination_${BuildConfig.MOENGAGE_SEGMENT_KOTLIN_VERSION}"
+
+        val mapper: Map<String, String> = mapOf(
+            USER_TRAIT_ANONYMOUS_ID to USER_ATTRIBUTE_SEGMENT_ID,
+            USER_TRAIT_EMAIL to USER_ATTRIBUTE_USER_EMAIL,
+            USER_TRAIT_UNIQUE_ID to USER_ATTRIBUTE_UNIQUE_ID,
+            USER_TRAIT_NAME to USER_ATTRIBUTE_USER_NAME,
+            USER_TRAIT_MOBILE to USER_ATTRIBUTE_USER_MOBILE,
+            USER_TRAIT_FIRST_NAME to USER_ATTRIBUTE_USER_FIRST_NAME,
+            USER_TRAIT_LAST_NAME to USER_ATTRIBUTE_USER_LAST_NAME,
+            USER_TRAIT_GENDER to USER_ATTRIBUTE_USER_GENDER,
+            USER_TRAIT_BIRTHDAY to USER_ATTRIBUTE_USER_BDAY
+
+        )
+    }
+
+    override val key: String
+        get() = MOENGAGE_SEGMENT_INTEGRATION_KEY
+
+    override fun update(settings: Settings, type: Plugin.UpdateType) {
+        try {
+            super.update(settings, type)
+            Logger.print { "$tag update(): will try to sync settings" }
+            if (type == Plugin.UpdateType.Initial) {
+                if (settings.hasIntegrationSettings(key)) {
+                    val moEngageSettings: MoEngageSettings =
+                        settings.destinationSettings(key) ?: return
+                    instanceId = moEngageSettings.apiKey
+                    integrationHelper =
+                        MoEIntegrationHelper(application, IntegrationPartner.SEGMENT)
+                    integrationHelper.initialize(instanceId, application)
+                    MoEIntegrationHelper.addIntegrationMeta(
+                        IntegrationMeta(
+                            INTEGRATION_META_TYPE,
+                            version()
+                        ), instanceId
+                    )
+                    Logger.print { "$tag update(): Segment Integration initialised." }
+                }
+                trackAnonymousId()
+            }
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag update(): " }
+        }
+
+    }
+
+    override fun alias(payload: AliasEvent): BaseEvent {
+        try {
+            super.alias(payload)
+            Logger.print { "$tag alias(): will try to update alias" }
+            MoEAnalyticsHelper.setAlias(application.applicationContext, payload.userId, instanceId)
+        } catch (t: Throwable) {
+            Logger.print(LogLevel.ERROR, t) { "$tag alias(): " }
+        }
+        return payload
+    }
+
+    override fun identify(payload: IdentifyEvent): BaseEvent {
+        try {
+            super.identify(payload)
+            Logger.print { "$tag identify(): will try to track IdentifyEvent" }
```

Comment:
log the payload if possible.

Reply:
done updated for rest of the required functions